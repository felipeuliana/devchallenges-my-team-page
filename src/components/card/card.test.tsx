import React from 'react'
import { render, screen } from '@testing-library/react'
import Card from './card'

const IMAGE = {
  alt: 'Woman avatar image.',
  src: 'michael-dam-mEZ3PoFGs_k-unsplash.jpg'
}
const TITLE = 'Art director'
const NAME = 'Saba Cabrera'

describe('Team member card', () => {
  beforeEach(() => {
    render(<Card image={IMAGE} name={NAME} title={TITLE} />)
  })

  it('should display the team member image', () => {
    const image = screen.getByRole('img')

    expect(image).toBeInTheDocument()
    expect(image).toHaveAttribute('alt', IMAGE.alt)
  })

  it('should display the team member title', () => {
    const title = screen.getByText(TITLE)

    expect(title).toBeInTheDocument()
  })

  it('should display the team member name', () => {
    const name = screen.getByText(NAME)

    expect(name).toBeInTheDocument()
  })
})
