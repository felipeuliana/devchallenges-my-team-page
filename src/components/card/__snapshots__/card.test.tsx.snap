// Vitest Snapshot v1

exports[`Card component > should render correctly 1`] = `
<div>
  <li
    class="_root_c37972"
  >
    <figure
      class="_figure_c37972"
    >
      <img
        alt="Woman avatar image."
        src="michael-dam-mEZ3PoFGs_k-unsplash.jpg"
      />
      <figcaption
        class="_caption_c37972"
      >
        Art director
      </figcaption>
    </figure>
    <h2
      class="_name_c37972"
    >
      Saba Cabrera
    </h2>
  </li>
</div>
`;

exports[`Team member card > should render correctly 1`] = `
<div>
  <li
    class="_root_c37972"
  >
    <figure
      class="_figure_c37972"
    >
      <img
        alt="Woman avatar image."
        src="michael-dam-mEZ3PoFGs_k-unsplash.jpg"
      />
      <figcaption
        class="_caption_c37972"
      >
        Art director
      </figcaption>
    </figure>
    <h2
      class="_name_c37972"
    >
      Saba Cabrera
    </h2>
  </li>
</div>
`;
